File: Tracks/csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static int squares = 64;
    public static ulong Square(int n)
    {
        if (n <= 0 || n > 64)
        {
            throw new ArgumentOutOfRangeException();
        }
        return (ulong)1 << (n - 1);
    }

    public static ulong Total() => Square(squares) * 2 - 1;
}

File: Tracks/typescript/grains/grains.ts
export default class Grains {
    static square = (tile: number, allowHigh: boolean = false): number => {
        if (tile < 1 || (tile > 64 && allowHigh === false)) {
            throw new Error("Tile out of range");
        }
        const index = tile - 1;
        switch (Math.sign(tile - 32) as 1 | 0 | -1) {
            case 1:
                return 2 ** index;
            case 0:
                return Math.abs(1 << index);
            case -1:
                return 1 << index;
        }
    };
    static total = () => Grains.square(65, true) - 1;
}
